### ISSUE TITLE
SSH modules broken after Net::SSH update

### ISSUE THREAD

**Issue by busterb** [rapid7/metasploit-framework#7321]
## Steps to reproduce

Run the following RC script, which iterates through many of the SSH-using modules, observe failures:

```
setg RHOSTS 127.0.0.1
setg RHOST 127.0.0.1
setg RPORT 22
setg USERNAME bcook
setg KEY_FILE ~/.ssh/authorized_keys

use auxiliary/scanner/ssh/apache_karaf_command_execution
run
use auxiliary/scanner/ssh/ssh_identify_pubkeys
run
use exploit/apple_ios/ssh/cydia_default_ssh
run
use exploit/linux/ssh/ceragon_fibeair_known_privkey
run
use exploit/linux/ssh/exagrid_known_privkey
run
use exploit/linux/ssh/f5_bigip_known_privkey
run
use exploit/linux/ssh/loadbalancerorg_enterprise_known_privkey
run
use exploit/linux/ssh/quantum_dxi_known_privkey
run
use exploit/linux/ssh/quantum_vmpro_backdoor
run
use exploit/linux/ssh/symantec_smg_ssh
run
use exploit/unix/ssh/array_vxag_vapv_privkey_privesc
run
use exploit/unix/ssh/tectia_passwd_changereq
run
```

```
resource (test.rc)> setg RHOSTS 127.0.0.1
RHOSTS => 127.0.0.1
resource (test.rc)> setg RHOST 127.0.0.1
RHOST => 127.0.0.1
resource (test.rc)> setg RPORT 22
RPORT => 22
resource (test.rc)> setg USERNAME bcook
USERNAME => bcook
resource (test.rc)> setg KEY_FILE ~/.ssh/authorized_keys
KEY_FILE => ~/.ssh/authorized_keys
resource (test.rc)> use auxiliary/scanner/ssh/apache_karaf_command_execution
resource (test.rc)> run
[*] 127.0.0.1:22 - Attempt to login...
[-] Auxiliary failed: ArgumentError invalid option(s): disable_agent, record_auth_info
[-] Call stack:
[-]   /Users/bcook/.rvm/gems/ruby-2.3.1@metasploit-framework/gems/net-ssh-3.2.0/lib/net/ssh.rb:204:in `start'
[-]   /Users/bcook/rapid7/metasploit-framework/modules/auxiliary/scanner/ssh/apache_karaf_command_execution.rb:89:in `block in do_login'
[-]   /Users/bcook/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/timeout.rb:91:in `block in timeout'
[-]   /Users/bcook/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/timeout.rb:33:in `block in catch'
[-]   /Users/bcook/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/timeout.rb:33:in `catch'
[-]   /Users/bcook/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/timeout.rb:33:in `catch'
[-]   /Users/bcook/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/timeout.rb:106:in `timeout'
[-]   /Users/bcook/rapid7/metasploit-framework/modules/auxiliary/scanner/ssh/apache_karaf_command_execution.rb:88:in `do_login'
[-]   /Users/bcook/rapid7/metasploit-framework/modules/auxiliary/scanner/ssh/apache_karaf_command_execution.rb:119:in `run_host'
[-]   /Users/bcook/rapid7/metasploit-framework/lib/msf/core/auxiliary/scanner.rb:134:in `block (2 levels) in run'
[-]   /Users/bcook/rapid7/metasploit-framework/lib/msf/core/thread_manager.rb:100:in `block in spawn'
[*] Auxiliary module execution completed
resource (test.rc)> use auxiliary/scanner/ssh/ssh_identify_pubkeys
resource (test.rc)> run
[-] 127.0.0.1:22 SSH - No valid keys found
[-] 127.0.0.1:22 SSH - No valid keys found
[-] 127.0.0.1:22 SSH - No valid cleartext keys found
[*] 127.0.0.1:22 SSH - Trying 0 cleartext key per user.
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
resource (test.rc)> use exploit/apple_ios/ssh/cydia_default_ssh
resource (test.rc)> run
[*] 127.0.0.1:22 - Attempt to login as 'root' with password 'alpine'
[-] Exploit failed: NameError undefined local variable or method `ssh_socket_factory' for #<Msf::Modules::Mod6578706c6f69742f6170706c655f696f732f7373682f63796469615f64656661756c745f737368::MetasploitModule:0x007f8194f78128>
[*] Exploit completed, but no session was created.
resource (test.rc)> use exploit/linux/ssh/ceragon_fibeair_known_privkey
resource (test.rc)> run
[-] Exploit failed: ArgumentError invalid option(s): msframework, msfmodule, record_auth_info
[*] Exploit completed, but no session was created.
resource (test.rc)> use exploit/linux/ssh/exagrid_known_privkey
resource (test.rc)> run
[-] Exploit failed: NameError undefined local variable or method `ssh_socket_factory' for #<Msf::Modules::Mod6578706c6f69742f6c696e75782f7373682f657861677269645f6b6e6f776e5f707269766b6579::MetasploitModule:0x007f8193f68c88>
Did you mean?  ssh_socket
[*] Exploit completed, but no session was created.
resource (test.rc)> use exploit/linux/ssh/f5_bigip_known_privkey
resource (test.rc)> run
[-] Exploit failed: ArgumentError invalid option(s): msframework, msfmodule, record_auth_info
[*] Exploit completed, but no session was created.
resource (test.rc)> use exploit/linux/ssh/loadbalancerorg_enterprise_known_privkey
resource (test.rc)> run
[-] Exploit failed: ArgumentError invalid option(s): record_auth_info
[*] Exploit completed, but no session was created.
resource (test.rc)> use exploit/linux/ssh/quantum_dxi_known_privkey
resource (test.rc)> run
[-] Exploit failed: ArgumentError invalid option(s): record_auth_info
[*] Exploit completed, but no session was created.
resource (test.rc)> use exploit/linux/ssh/quantum_vmpro_backdoor
resource (test.rc)> run
[*] 127.0.0.1:22 - Attempt to login...
[-] Exploit failed: ArgumentError invalid option(s): record_auth_info
[*] Exploit completed, but no session was created.
resource (test.rc)> use exploit/linux/ssh/symantec_smg_ssh
resource (test.rc)> run
[*] 127.0.0.1:22 - Attempt to login...
[-] Exploit failed: ArgumentError invalid option(s): record_auth_info
[*] Exploit completed, but no session was created.
resource (test.rc)> use exploit/unix/ssh/array_vxag_vapv_privkey_privesc
resource (test.rc)> run
[*] Started reverse TCP double handler on 127.0.0.1:4444
[*] 127.0.0.1:22 - Attempt to login with 'sync:SSH PRIVATE KEY'
[-] Exploit failed: ArgumentError invalid option(s): record_auth_info
[*] Exploit completed, but no session was created.
resource (test.rc)> use exploit/unix/ssh/tectia_passwd_changereq
resource (test.rc)> run
[-] 127.0.0.1:22 - 127.0.0.1:22 - This host does not use password method authentication
[*] Exploit completed, but no session was created.
```

## Expected behavior
The modules should largely fail, but not due to typos, missing requires, invalid arguments, missing functionality, etc.

What should happen?

## Current behavior

What happens instead?
Many of the modules fail for reasons unrelated to the lack of a suitable target.

### Metasploit version

Latest git checkout, OS X.

**Comment by busterb** [comment]
I have partial fixes for some of these modules, but others require deeper rework of the modules themselves. Will submit partial PR in a bit.

**Comment by bcook-r7** [comment]
WIP branch: https://github.com/bcook-r7/metasploit-framework/tree/fix-ssh-modules

**Comment by h00die** [comment]
+1 on this, replicated exploit/linux/ssh/exagrid_known_privkey
[-] Exploit failed: NameError undefined local variable or method `ssh_socket_factory'

**Comment by bcook-r7** [comment]
Fixed in #7339
